src/domain/settlement/index.js

```diff
@@ -47,10 +44,13 @@ const createSettlementModel = async (settlementModel, trx = null) => {
       settlementModel.requireLiquidityCheck,
       ledgerAccountType.ledgerAccountTypeId, settlementAccountType.ledgerAccountTypeId, settlementModel.autoPositionReset, trx)
 
-    // create the accounts required for the settlementModel for existing participants
-    await ParticipantService.createAssociatedParticipantAccounts(settlementModel.currency, ledgerAccountType.ledgerAccountTypeId, trx)
-    await ParticipantService.createAssociatedParticipantAccounts(settlementModel.currency, settlementAccountType.ledgerAccountTypeId, trx)
-
+    if (settlementModel.currency) {
+      // check for existing hub account with the settlementModel to be able to create participant accounts automatically
+      await ParticipantService.validateHubAccounts(settlementModel.currency)
+      // create the accounts required for the settlementModel for existing participants
+      await ParticipantService.createAssociatedParticipantAccounts(settlementModel.currency, ledgerAccountType.ledgerAccountTypeId, trx)
```

Comment:
we can think about it, yeah. Although aren't we mixing some of the workflows? To create an account during transfer process is not exactly straightforward but on the other hand, if this FSP doesn't trade never in that currency, maybe is worth it.

Reply:
2 questions:

1. This (current implementation, not Miguel's suggestion) feels like an odd side effect to me. If a user is creating a settlement model, and the ledger account type in their specified currency doesn't exist, shouldn't that be an error?

2. It also seems odd to me that by specifying the currency in this request (which users will default to because it was previously a required property) a bunch of other stuff gets created. Shouldn't the user explicitly have to create the 'associated participant accounts' themselves?